### ISSUE TITLE
parse types in Style

### ISSUE THREAD

**Issue by hypotext** [penrose/penrose#408]
Request from @keenancrane:

Can we parse type keywords in front of declarations? (But not necessarily do typechecking.) I thought this already worked in the parser, but we just tried it in the most recent commit on `hyperbolic`, in `hyperbolic-domain/PoincareDisk.sty` and it didn't work:

`vec2 u = p.dotShape.center / H.diskShape.r ` (added type keyword `vec2` on line 48)

```
penrose: Error when parsing Style:  48:4:    
| 48 |    vec2 u = p.dotShape.center / H.diskShape.r    
|    ^ unexpected 'v' expecting '}' 
```

@wodeni Would you mind looking into the parser to see if this could be quickly doable?

**Comment by wodeni** [comment]
Type annotation is currently not parsed due to the issue discussed [here](https://github.com/penrose/penrose/pull/398#issuecomment-726225024).

**Comment by hypotext** [comment]
Fixed in #411
